src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,556 @@
+# Copyright (c) 2020: Andrew Rosemberg and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    fill_off_diagonal(H)
+
+Filling the off-diagonal elements of a sparse matrix to make it symmetric.
+"""
+function fill_off_diagonal(H)
+    ret = H + H'
```

Comment:
Do we assume `H` is lower-triangular?

Reply:
I was following Oscar's steps from here: 

https://github.com/jump-dev/JuMP.jl/blob/54c1239ee964b13012a4553b4e36bfb5579e6aff/docs/src/tutorials/nonlinear/querying_hessians.jl#L204-L221